frontend/src/component/releases/ReleasePlanTemplate/MilestoneCard.tsx

```diff
@@ -0,0 +1,109 @@
+    marginTop: theme.spacing(-0.25),
+    marginLeft: theme.spacing(0.5),
+    height: theme.spacing(2.5),
+    width: theme.spacing(2.5),
+    color: theme.palette.text.secondary,
+}));
+
+const StyledGridItem = styled(Grid)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'center',
+}));
+
+const StyledInput = styled(Input)(({ theme }) => ({
+    width: '100%',
+}));
+
+const StyledMilestoneCardTitle = styled('span')(({ theme }) => ({
+    fontWeight: theme.fontWeight.bold,
+    fontSize: theme.fontSizes.bodySize,
+}));
+
+interface IMilestoneCard {
+    index: number;
+    milestone: IReleasePlanMilestonePayload;
+    showAddStrategyDialog: (index: number) => void;
+    errors: { [key: string]: string };
+    clearErrors: () => void;
+}
+
+export const MilestoneCard = ({
+    index,
+    milestone,
+    showAddStrategyDialog,
+    errors,
+    clearErrors,
+}: IMilestoneCard) => {
+    const [milestoneName, setMilestoneName] = useState<string>(milestone.name);
+    const [editMode, setEditMode] = useState<boolean>(false);
+    const theme = useTheme();
+
+    useEffect(() => {
+        milestone.name = milestoneName;
+    }, [milestoneName]);
+
+    const editMilestoneNameClick = () => {
+        setEditMode(true);
+    };
+
+    return (
+        <StyledMilestoneCard>
+            <StyledMilestoneCardBody>
+                <Grid container>
+                    <StyledGridItem item xs={10} md={10}>
+                        {editMode && (
+                            <StyledInput
+                                label=''
+                                value={milestoneName}
+                                onChange={(e) =>
+                                    setMilestoneName(e.target.value)
+                                }
+                                error={Boolean(errors?.name)}
+                                errorText={errors?.name}
+                                onFocus={() => clearErrors()}
+                                onBlur={() => setEditMode(false)}
+                                autoFocus
+                                onKeyDownCapture={(e) => {
+                                    if (e.code === 'Enter') {
+                                        e.preventDefault();
+                                        e.stopPropagation();
+                                        setEditMode(false);
+                                    }
+                                }}
+                            />
+                        )}
+                        {!editMode && (
+                            <StyledMilestoneCardTitle
+                                onClick={editMilestoneNameClick}
+                            >
+                                {milestoneName}
+                            </StyledMilestoneCardTitle>
+                        )}
+                        {!editMode && (
+                            <StyledEditIcon onClick={editMilestoneNameClick} />
+                        )}
```

Comment:
Why isn't this a part of the above block, since the condition is the same?

Reply:
Argh, I moved things around and forgot to remove the duplicate condition 🙄

Reply:
I think you could have added the onClick handler on the parent component, catching any click in the children, instead of duplicating it.

Reply:
Parent is the same for this and the other section which is only shown during edit mode, and would have a weird effect if you click inside. I'll have a think if I can introduce a parent just for these two while doing add strategies